sectionproperties/pre/sections.py

```diff
@@ -534,6 +534,142 @@ def __init__(self, d, n, shift=[0, 0]):
         self.shift_section()
 
 
+class EllipticalSection(Geometry):
+    """Constructs a solid ellipse centered at the origin *(0, 0)* with vertical diameter
+    *d_y* and horizontal diameter *d_x*, using *n* points to construct the ellipse.
+
+    :param float d_y: Diameter of the ellipse in the y-dimension
+    :param float d_x: Diameter of the ellipse in the x-dimension
+    :param int n: Number of points discretising the ellipse
+    :param shift: Vector that shifts the cross-section by *(x, y)*
+    :type shift: list[float, float]
+
+    The following example creates an elliptical cross-section with a vertical diameter of
+    50 and horizontal diameter of 25, with 40 points, and generates a mesh with a maximum triangular area of
+    2.5:
+
+        import sectionproperties.pre.sections as sections
+
+        geometry = sections.EllipticalSection(d_y=50, d_x=25, n=40)
+        mesh = geometry.create_mesh(mesh_sizes=[2.5])
+
+    ..  figure:: ../images/sections/ellipse_geometry.png
+        :align: center
+        :scale: 75 %
+
+        Elliptical section geometry.
+
+    ..  figure:: ../images/sections/ellipse_mesh.png
+        :align: center
+        :scale: 75 %
+
+        Mesh generated from the above geometry.
+    """
+
+    def __init__(self, d_y, d_x, n, shift=[0, 0]):
+        """Inits the EllipticalSection class."""
+
+        # assign control point
+        control_points = [[0, 0]]
+
+        super().__init__(control_points, shift)
+
+        # loop through each point on the ellipse
+        for i in range(n):
+            # determine polar angle
+            theta = i * 2 * np.pi * 1.0 / n
+
+            # calculate location of the point
+            x = 0.5 * d_x * np.cos(theta)
+            y = 0.5 * d_y * np.sin(theta)
+
+            # append the current point to the points list
+            self.points.append([x, y])
+
+            # if we are not at the last point
+            if i != n - 1:
+                self.facets.append([i, i + 1])
+            # if we are at the last point, complete the ellipse
+            else:
+                self.facets.append([i, 0])
+
+        self.shift_section()
+
+
+class Ehs(Geometry):
+    """Constructs an elliptical hollow section centered at the origin *(0, 0)*,
+    with outer vertical diameter *d_yo*, outer horizontal diameter *d_xo*, and 
+    thickness *t*, using *n* points to construct the inner and outer ellipses. 
+    Note that the thickness of a hollow ellipse does not stay constant all
```

Comment:
Hi Benjamin

I wonder if you can expand on this, it's my understanding that the perpendicular to surface through thickness is constant. I checked your code against the British EHS's and it agreed quite well so I assume your formulation is correct as it appears to create a section that has a constant through thickness based on plots. So the note saying the thickness is not constant was a little confusing to me. Can you elaborate on this if the through thickness is not constant. 

These were the only commercial EHS I could find properties for
[https://www.steelconstruction.info/images/e/eb/Celsius_355_NH_Technical_Guide_2018.pdf](https://www.steelconstruction.info/images/e/eb/Celsius_355_NH_Technical_Guide_2018.pdf)

Reply:
Thanks Ben for the useful additions to the sections file! And thanks Agent for the proofing!